nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-rest-api/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/comment/CommentAdapter.java

```diff
@@ -79,6 +89,10 @@ public Comment getExternalComment(@PathParam("entityId") String entityId) {
     @PUT
     @Path("{commentId}")
     public Comment updateComment(@PathParam("commentId") String commentId, Comment comment) {
+        if (!getContext().getCoreSession().exists(new IdRef(commentId))) {
+            throw new DocumentNotFoundException("The comment " + commentId + " does not exist.");
+        }
+
```

Comment:
Flo will probably tell you to remove this extra line :D

Reply:
The service is actually checking that the document exists but throw an `IllegalArgumentException` why doing this extra check?  If for a good reason, then the service should throw `DocumentNotFoundException` instead

Reply:
I agree with you but I think @funshodavid may have a strong argument for that.

Reply:
Strictly speaking `DocumentNotFoundException` is an implementation detail of how comments are stored. They may not be stored as documents so it doesn't really make sense to throw that.
@funshodavid plans on making changes to the comment service itself to throw a `NuxeoException` with a 400 status code instead of an `IllegalArgumentException`. Would a 400 be ok for the client part @nmpcunha?